### PR TITLE
CRM: Resolves #3323 - implement task entity

### PR DESCRIPTION
**Pull request by tbradsha**
CRM: Resolves Automattic/zero-bs-crm#3323 - implement task entity

## Proposed changes:

This implements a basic Task entity object.

Note that the original issue mentioned Events, but I went with Tasks, as that's what we use throughout the UI, and also "events" could get confusing when dealing with the event manager system:

* `id`: the ID in the table
* `owner`: the WP user ID for the user that owns the task
* `title`: the task title
* `description`: the task description
* `start`: Unix timestamp of task start
* `end`: Unix timestamp of task end
* `is_completed`: Whether task has been completed or not
* `show_in_portal`: Whether task should show in portal
* `show_in_calendar`: Whether task should show in calendar
* `created`: Unix timestamp of task creation
* `lastupdated`: Unix timestamp of task last update

UTS fields are initially set to `null`, as `-1` is a valid UTS timestamp.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

N/A at the moment.

### CODE REVIEW

**Comment by cleacos** [comment]
In the backend, let's keep the same name we already have for it, `event`. The current model with the DAL, and the current Automation code use `event`. In the UI we could use any name that fits better for the end user. I prefer the name Task everywhere, but at least let's keep the same name in the backend.

File: projects/plugins/crm/src/entities/class-task.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * Task Entity.
+ *
+ * @package automattic/jetpack-crm
+ */
+
+namespace Automattic\Jetpack\CRM\Entities;
+
+/**
+ * Task class.
```
Comment (cleacos) [inline projects/plugins/crm/src/entities/class-task.php:11]: We should comment here about the relationship with the `events` on the DAL (events endpoint) and DB (events table) sides.

Comment (tbradsha) [reply projects/plugins/crm/src/entities/class-task.php:11]: 3b36c25d3e

**Review by cleacos — APPROVED** [review]
LGTM!

**Pull request merged**